core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -2751,17 +2746,141 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
     assert(failedStages.collect {
       case stage: ResultStage => stage
     }.head.findMissingPartitions() == Seq(0, 1))
-
     scheduler.resubmitFailedStages()
+    (shuffleId1, shuffleId2)
+  }
+
+  test("SPARK-25341: abort stage while using old fetch protocol") {
+    // reset the test context with using old fetch protocol
+    afterEach()
+    val conf = new SparkConf()
+    conf.set(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL.key, "true")
+    init(conf)
+
+    val (shuffleId1, _) = constructIndeterminateStageRetryScenario()
+    // The second task of the `shuffleMapRdd2` failed with fetch failure
```

Comment:
This is a little misleading. `shuffleMapRdd2` only miss its first partition, it's weird to have the second task fail. Can we fail the first task?

Reply:
Great thanks for the careful review, change to fail the first task in 7259500.